tests/zfs-tests/tests/functional/cli_root/zhack/library.kshlib

```diff
@@ -0,0 +1,293 @@
+#!/bin/ksh
+
+#
+# This file and its contents are supplied under the terms of the
+# Common Development and Distribution License ("CDDL"), version 1.0.
+# You may only use this file in accordance with the terms of version
+# 1.0 of the CDDL.
+#
+# A full copy of the text of the CDDL should have accompanied this
+# source.  A copy of the CDDL is also available via the Internet at
+# http://www.illumos.org/license/CDDL.
+#
+
+#
+# Copyright (c) 2021 by vStack. All rights reserved.
+#
+
+. "$STF_SUITE"/include/libtest.shlib
+. "$STF_SUITE"/include/blkdev.shlib
+
+#
+# Description:
+#
+# Test whether zhack label repair commands can recover detached devices
+# and corrupted checksums with a variety of sizes, and ensure
+# the purposes of either command is cleanly separated from the others.
+#
+# Strategy:
+#
+# Tests are done on loopback devices with sizes divisible by label size and sizes that are not.
```

Comment:
Is it really necessary to do the even/odd tests with every test type, or should it be sufficient with only one. Realistically, only a very small section of shared code is being tested by doing this.

Reply:
My feeling is it's not really buying us much.  But if we wanted to maintain the test coverage you could do something like half the time use MINVDEVSIZE and the other half MINVDEVSIZE + random(MINVDEVSIZE).  This way we'd be testing both regularly and should notice quickly any problem in the CI.

Reply:
> or a randomly determined choice between those two options

This is what I had in mind, even though it makes the test not 100% deterministic.  The behavior it's testing must still always work regardless of the exact vdev size.

Reply:
I have an idea to benefit from random testing, while ensuring all necessary cases are tested at least once. As I understand, fundamentally, we want to test unexpected random cases, which are divisible by the label size, and cases which are not.

We keep track of the number of remaining tests, and the number of the latter two cases that still need to be tested. Starting by always generating a random size, could determine whether it matches either of the latter two conditions, decrementing the count of them, and setting a flag to represent either. When the number of remaining tests matches the number of remaining cases, then we switch to static sizes.

Reply:
Ended up just doing it more or less the way you described, because what I described would involve making each individual test non-separate.